src/Streamly/Coreutils/Cp.hs

```diff
@@ -0,0 +1,281 @@
+module Streamly.Coreutils.Cp
+    ( CpOptions (..)
+    , defaultCpOptions
+    , cpFileWithRename
+    , cpFiles
+    , traverseDir
+    , cpDir
+   )
+where
+
+import Path
+import Streamly
+import Streamly.Coreutils.Common
+
+import qualified Streamly.Prelude as S
+import qualified Streamly.Internal.Prelude as IP
+import qualified Streamly.Internal.FileSystem.File as File
+
+import System.Info (os)
+import System.IO.Unsafe (unsafePerformIO)
+import System.Directory (listDirectory, doesFileExist, createDirectoryIfMissing)
+
+-------------------------------------------------------------------------------
+-- Record for options used with cp
+-------------------------------------------------------------------------------
+
+-- | Data type to represent flags\/options in GNU @cp@
+--
+-- @since 0.1.0.0
+data CpOptions = CpOptions
+    { verbose :: Bool
+    -- ^ Prints the source and destination files being copied
+    , force :: Bool
+    -- ^ if the destination file
+    -- cannot be opened, removes it and continues copying the rest
+    , overwriteExisting :: Bool
+    -- ^ If @True@, an existing destination file will be overwritten
+    -- due to copying
+    }
+
+
+-- | Default options for @cp@ are @verbose@ and @overwriteExisting@ to
+-- @True@ and @force@ set to @False@
+--
+-- @since 0.1.0.0
+{-# INLINE defaultCpOptions #-}
+defaultCpOptions :: CpOptions
+defaultCpOptions = CpOptions True False True
+
+
+-------------------------------------------------------------------------------
+-- cp and helper functions for options
+-------------------------------------------------------------------------------
+
+
+-- | Basic file copying with @FilePath@ as argument types.
+-- Prints paths of source, destination files when @verbose@ is @True@.
+-- Returns @True@ if copying operation was successful.
+-- @False@ is returned in cases where a file by the same name
+-- exists in the destination path and @overwriteExisting@ is set to
+-- @False@ or if the source file does not exist.
+--
+-- @since 0.1.0.0
+{-# INLINE copy #-}
+copy
+    :: CpOptions
+    -> FilePath
+    -> FilePath
+    -> IO Bool
+copy opt src dst = do
+    existDst <- doesFileExist dst
+    existSrc <- doesFileExist src
+    if not existSrc
```

Comment:
if/else indentation

Reply:
If the options are exclusive, use `when` instead.

Reply:
https://github.com/composewell/haskell-dev/blob/master/coding-style.rst#if-then-else
What is the issue with this indentation style?